src/components/resource/_fields/resource-coverage-fields/resource-coverage-fields.js

```diff
@@ -0,0 +1,196 @@
+import React, {
+  Component,
+  Fragment
+} from 'react';
+
+import PropTypes from 'prop-types';
+import { Field } from 'react-final-form';
+import { FieldArray } from 'react-final-form-arrays';
+import {
+  FormattedMessage,
+  injectIntl,
+  intlShape,
+} from 'react-intl';
+
+import moment from 'moment';
+import isEqual from 'lodash/isEqual';
+
+import {
+  Icon,
+  RepeatableField,
+  Datepicker,
+  RadioButton,
+} from '@folio/stripes/components';
+
+import CoverageDateList from '../../../coverage-date-list';
+
+import validateDateRange from '../validate-date-range';
+import { isBookPublicationType } from '../../../utilities';
+
+import styles from './resource-coverage-fields.css';
+
+class ResourceCoverageFields extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    model: PropTypes.object.isRequired,
+  };
+
+  validateDateRange = (values) => {
+    const {
+      intl: { locale },
+      model: {
+        package: {
+          customCoverage: packageDateRange
+        },
+      },
+    } = this.props;
+
+    return validateDateRange(values, locale, packageDateRange);
+  }
+
+  renderCustomCoverageFields = ({ fields, name: fieldsName, meta: { initial: initialValues } }) => {
```

Comment:
I think it will be better to do destructurisation inside function body with normal formatting